### ISSUE TITLE
RTL languages support in CP

### ISSUE THREAD

**Issue by Krzemo** [statamic/ideas#248]
Not having appropriate dir set for RTL languages cases issues with saved content.

I think title is pretty self explanatory.
config/statamic/sites.php could look like this:
    'sites' => [

        'en' => [
            'name' => 'English',
            'locale' => 'en_US',
            'url' => '/',
        ],

        'ar' => [
            'name' => 'عربي',
            'locale' => 'ar_SA',
            'url' => '/ar/',
            'dir' => 'rtl',
        ],

    ]

This would add dir attr to opening <html> tag:
`<html lang="ar-ar" dir="rtl">`

Plus tons of work to make CP accomodate this.

**Comment by tao** [comment]
I also require RTL support on the Control Panel. I'm working with Hebrew that is RTL.

![Screenshot 2021-09-20 at 13 09 25](https://user-images.githubusercontent.com/1446331/133993353-de16a6ac-6c45-49fc-986f-6abc8fdd8b79.png)

It should be right aligned text, but it seems like the cursor also behaves strangely in RTL mode with the markdown parser. It's not really detecting well where the end of the line is and keeps putting the full-stop on the next line.

![Screenshot 2021-09-20 at 13 10 50](https://user-images.githubusercontent.com/1446331/133993376-f374cf73-8b7d-4899-9ede-3a3b8f0afd9d.png)

On my actual website design I have to flip the entire site horizontally for the Hebrew readers, but on the Control Panel we could get away with keeping the backend LTR like the rest of the languages, but the content fields need a bit of love.

**Comment by grantholle** [comment]
Am I missing something on how to get this going? I have a `'direction' => 'rtl'` entry on my site config but the CP still is LTR. I like the solution that @tao has for the class to just make the input RTL.

**Comment by jasonvarga** [comment]
Adding `'direction' => 'rtl'` works without a problem for me.

Can you show your `config/statamic/sites.php` file?

**Comment by grantholle** [comment]
Yeah that's right. If you check out what @tao is talking about above, it's fine that the CP is LTR, except if I'm editing a page (or whatever) for a site that has `'direction' => 'rtl'`, the _input_ should, at the very least, be RTL. His class-based approach is simple and clever and gets the job done.

**Comment by tao** [comment]
Yeah, it works now if you select `Hebrew` at the top of the site for example:

![Screenshot 2022-06-04 at 14 22 53](https://user-images.githubusercontent.com/1446331/171998775-0d7f62ab-d396-405d-b412-88031065d283.png)

The control panel is all messed up but I don't really mind that. The input works and we can fix the style of the control panel later by updating the CSS... now that the `dir=rtl` in the html we can target that and start making an RTL version of the control panel.

@jasonvarga what we were talking about was just changing the direction of the input if the site is selected in the sidebar but the site picker is still English:

![Screenshot 2022-06-04 at 14 17 44](https://user-images.githubusercontent.com/1446331/171998603-125c1d7d-1ebe-4f64-9320-031ed8c6b128.png)

If you just add `dir=rtl` on the tab panel then it gives the input fields in the correct direction even if the site selector is in English.

**Comment by tao** [comment]
If we can add something like this to the `layout.blade.php` file then we can achieve this:

```php
<div class="workspace">
    <div class="page-wrapper page-wrapper-{{ $entry->site()->direction() }}" :class="wrapperClass">
        @yield('content')
    </div>
</div>
```

That will give us this in the css

```html
<div class="workspace">
    <div class="page-wrapper page-wrapper-rtl max-w-3xl">
         @yield('content')
    </div>
</div>
```

Then in `cp.css` I can add this line:

```css
.page-wrapper-rtl .publish-section-wrapper {
    direction: rtl;
}
```

And it works:

![Screenshot 2022-06-04 at 15 14 15](https://user-images.githubusercontent.com/1446331/172000659-839009a8-f20c-440c-af35-da93371319c6.png)
